zap/src/main/java/org/parosproxy/paros/extension/history/ExtensionHistory.java

```diff
@@ -463,13 +464,19 @@ public void run() {
         }
     }
 
-    private void searchHistory(HistoryFilter historyFilter) {
+    /**
+     * Returns a sorted list of History IDs for the visible history references.
+     *
+     * @return a list of History IDs for the visible history references.
+     * @since 2.13.0
+     */
+    public List<Integer> getHistoryIds() {
         Session session = getModel().getSession();
+        List<Integer> list;
```

Comment:
This change is not necessary, could just return the result too.

Reply:
Should be inside the try :wink: Also synchronize is no longer necessary here.